assets/src/edit-story/app/story/actions/useSaveStory.js

```diff
@@ -25,19 +25,21 @@ import { renderToStaticMarkup } from 'react-dom/server';
  */
 import addQueryArgs from '../../../utils/addQueryArgs';
 import { useAPI } from '../../api';
-import { OutputPage } from '../../../output';
+import { useConfig } from '../../config';
+import OutputStory from '../../../output/story';
 
 /**
  * Creates AMP HTML markup for saving to DB for rendering in the FE.
  *
- * @param {Object} pages Object of pages.
- * @return {Element} Markup of pages.
+ * @param {import('../../../types').Story} story Story object.
+ * @param {Array<Object>} pages List of pages.
+ * @param {Object} metadata Metadata.
+ * @return {Element} Story markup.
  */
-const getStoryMarkupFromPages = (pages) => {
-  const markup = pages.map((page) => {
-    return renderToStaticMarkup(<OutputPage page={page} />);
-  });
-  return markup.join('');
+const getStoryMarkup = (story, pages, metadata) => {
+  return renderToStaticMarkup(
```

Comment:
I keep reading that `react-dom/server` is rather expensive on client-side. Any truth to that?

Reply:
Not enough experience with it to be honest, but it doesn‘t seem like it so far. Perhaps @barklund has some thoughts?

Reply:
I have no relevant opinion or experience on this. Sorry 😞.

Reply:
What does the bundle size plugin says?

Reply:
It was introduced as part of #326, which says roughly 7KB (there were some other additions there too though)

Reply:
Ok. Let's ignore this then.